src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -310,19 +312,14 @@ private static IntPtr GetSslMethod(bool isServer, long options)
             return method;
         }
 
-        //TODO See if SSL_CTX_Set_quite_shutdown can be used
         private static void Disconnect(IntPtr sslPtr)
         {
             if (IntPtr.Zero != sslPtr)
             {
                 int retVal = libssl.SSL_shutdown(sslPtr);
-                if (0 == retVal)
+                if (retVal < 0)
                 {
-                    retVal = libssl.SSL_shutdown(sslPtr);
-                    if (retVal < 0)
-                    {
-                        libssl.SSL_get_error(sslPtr, retVal);
-                    }
+                    libssl.SSL_get_error(sslPtr, retVal);
```

Comment:
I think there is an error queue but it is specific to sslPtr. Since that is getting freed as the next step, this should be ok.
But even otherwise, if the error is left in the queue, the queue is supposed to be consulted only if some SSL API fails in which case the new error would be at the front of the queue, right?